load_balancer.tf

```diff
@@ -29,16 +29,34 @@ resource "aws_lb" "public_lb" {
 
 resource "aws_lb_target_group" "ec2_tg" {
   name     = "Terraform-EC2-TargetGroup"
-  port     = 80
-  protocol = "HTTP"
+  port     = 443
+  protocol = "HTTPS"
   vpc_id   = "${aws_vpc.main.id}"
 }
 
 resource "aws_lb_listener" "port_80_listener" {
   load_balancer_arn = "${aws_lb.public_lb.arn}"
-  port              = "80"
+  port              = 80
   protocol          = "HTTP"
 
+  default_action {
+    type = "redirect"
+
+    redirect {
+      port        = 443
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
+
+resource "aws_lb_listener" "port_443_listener" {
+  load_balancer_arn = "${aws_lb.public_lb.arn}"
+  port              = 443
+  protocol          = "HTTPS"
+  ssl_policy        = "TODO"
+  certificate_arn   = "TODO"
```

Comment:
For setting this up we could upload the SSL certificate in the [Certificate Manager](https://eu-west-1.console.aws.amazon.com/acm/home?region=eu-west-1#/) through the AWS Dashboard, and then use a [Data source](https://www.terraform.io/docs/providers/aws/d/acm_certificate.html) to reference the Certificate inside terraform. For more info on Data Sources see [this](https://www.terraform.io/docs/configuration/data-sources.html).